apis/v1alpha1/cluster_bpf_application_state_types.go

```diff
@@ -24,89 +24,89 @@ import (
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Kprobe' ?  has(self.kprobeInfo) : !has(self.kprobeInfo)",message="kprobeInfo configuration is required when type is kprobe, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'KretProbe' ?  has(self.kretprobeInfo) : !has(self.kretprobeInfo)",message="kretprobeInfo configuration is required when type is kretprobe, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Uprobe' ?  has(self.uprobeInfo) : !has(self.uprobeInfo)",message="uprobeInfo configuration is required when type is uprobe, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'UretProbe' ?  has(self.uretprobeInfo) : !has(self.uretprobeInfo)",message="uretprobeInfo configuration is required when type is uretprobe, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'TracePoint' ?  has(self.tracepointInfo) : !has(self.tracepointInfo)",message="tracepointInfo configuration is required when type is tracepoint, and forbidden otherwise"
 type ClBpfApplicationProgramState struct {
 	BpfProgramStateCommon `json:",inline"`
-	// Type specifies the bpf program type
+	// type specifies the bpf program type
 	// +unionDiscriminator
-	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:Enum:="xdp";"tc";"tcx";"fentry";"fexit";"kprobe";"kretprobe";"uprobe";"uretprobe";"tracepoint"
+	// +required
+	// +kubebuilder:validation:Enum:="XDP";"TC";"TCX";"Fentry";"Fexit";"Kprobe";"KretProbe";"Uprobe";"UretProbe";"TracePoint"
 
-	Type EBPFProgType `json:"type,omitempty"`
+	Type EBPFProgType `json:"type"`
 
-	// xdp defines the desired state of the application's XdpPrograms.
+	// xdpInfo defines the desired state of the application's XdpPrograms.
 	// +unionMember
 	// +optional
 	XDPInfo *ClXdpProgramInfoState `json:"xdpInfo,omitempty"`
 
-	// tc defines the desired state of the application's TcPrograms.
+	// tcInfo defines the desired state of the application's TcPrograms.
 	// +unionMember
 	// +optional
 	TCInfo *ClTcProgramInfoState `json:"tcInfo,omitempty"`
 
-	// tcx defines the desired state of the application's TcxPrograms.
+	// tcxInfo defines the desired state of the application's TcxPrograms.
 	// +unionMember
 	// +optional
 	TCXInfo *ClTcxProgramInfoState `json:"tcxInfo,omitempty"`
 
-	// fentry defines the desired state of the application's FentryPrograms.
+	// fentryInfo defines the desired state of the application's FentryPrograms.
 	// +unionMember
 	// +optional
 	FentryInfo *ClFentryProgramInfoState `json:"fentryInfo,omitempty"`
 
-	// fexit defines the desired state of the application's FexitPrograms.
+	// fexitInfo defines the desired state of the application's FexitPrograms.
```

Comment:
I guess that would become `fExitInfo` 👀 ?

Reply:
no more info